### PR TITLE
Problem: missing api for new_wallet, approve_for_all etc. (fix #105)

### LINKED ISSUE #105: Problem: missing api for new_wallet, approve_for_all etc.
add missing apis for defi-wallet-core

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add more apis

### CODE REVIEW

**Comment by ricky321q** [comment]
So we also have to change blue print classes?
Btw Can't find erc20transferfrom?

**Comment by leejw51crypto** [comment]
it's already existed
https://github.com/cronos-labs/play-unreal-plugin/blob/main/Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp

in this pr, also added remaining c++ nft apis

**Review by tomtau — COMMENTED** [review]
given this is _insecure_, this should IMHO be indicated in the names, e.g. `InitializeNewDevelopmentOnlyWallet`

**Comment by leejw51crypto** [comment]
ok, i think so.  i'll change

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp [hunk 1]
```diff
@@ -1086,6 +1331,91 @@ void ADefiWalletCoreActor::Erc721TransferFrom(
...
+    if (NULL == _coreWallet) {
+      success = false;
+      output_message = TEXT("Invalid Wallet");
+      return;
+    }
+
+    char hdpath[100];
+    snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/%d", EthCoinType,
+             walletindex);
+    rust::cxxbridge1::Box<PrivateKey> privatekey = _coreWallet->get_key(hdpath);
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:1387]: could this be generic over the wallet -- i.e. not hardcode the development wallet, but work for any wallet method (WalletConnect, Web3Auth, Magic/Fortmatic...)??

**Review by tomtau — APPROVED** [review]
could the methods be generic?

**Review by damoncro — APPROVED** [review]

**Comment by damoncro** [comment]
Make sure it can be compiled under RunUAT

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 1 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:1387]: yes, possible, 
https://github.com/crypto-com/defi-wallet-core-rs/issues/453
i'll apply generic signing for (walletconnect, various wallets etc)

**Pull request merged**